tests/Comparator/Int64ComparatorTest.php

```diff
@@ -0,0 +1,211 @@
+            'actualValue' => new Int64(123),
+        ];
+
+        yield 'Expects Int64, Actual int' => [
+            'expectedResult' => true,
+            'expectedValue' => new Int64(123),
+            'actualValue' => 123,
+        ];
+
+        yield 'Expects Int64, Actual int string' => [
+            'expectedResult' => true,
+            'expectedValue' => new Int64(123),
+            'actualValue' => '123',
+        ];
+
+        yield 'Expects Int64, Actual float' => [
+            'expectedResult' => true,
+            'expectedValue' => new Int64(123),
+            'actualValue' => 123.0,
+        ];
+
+        yield 'Expects Int64, Actual float string' => [
+            'expectedResult' => true,
+            'expectedValue' => new Int64(123),
+            'actualValue' => '123.0',
+        ];
+
+        yield 'Expects Int64, Actual non-numeric string' => [
+            'expectedResult' => false,
+            'expectedValue' => new Int64(123),
+            'actualValue' => 'foo',
+        ];
+
+        yield 'Expects int, Actual Int64' => [
+            'expectedResult' => true,
+            'expectedValue' => 123,
+            'actualValue' => new Int64(123),
+        ];
+
+        yield 'Expects int string, Actual Int64' => [
+            'expectedResult' => true,
+            'expectedValue' => '123',
+            'actualValue' => new Int64(123),
+        ];
+
+        yield 'Expects float, Actual Int64' => [
+            'expectedResult' => true,
+            'expectedValue' => 123.0,
+            'actualValue' => new Int64(123),
+        ];
+
+        yield 'Expects float string, Actual Int64' => [
+            'expectedResult' => true,
+            'expectedValue' => '123.0',
+            'actualValue' => new Int64(123),
+        ];
+
+        yield 'Expects non-numeric string, Actual Int64' => [
+            'expectedResult' => false,
+            'expectedValue' => 'foo',
+            'actualValue' => new Int64(123),
+        ];
+
+        yield 'Expects float, Actual Float' => [
+            'expectedResult' => false,
+            'expectedValue' => 123.0,
+            'actualValue' => 123.0,
+        ];
+
+        yield 'Expects numeric string, Actual numeric string' => [
+            'expectedResult' => false,
+            'expectedValue' => '123',
+            'actualValue' => '123',
+        ];
+    }
+
+    /**
+     * @dataProvider provideMatchingAssertions
+     * @doesNotPerformAssertions
+     */
+    public function testMatchingAssertions($expected, $actual): void
+    {
+        (new Int64Comparator())->assertEquals($expected, $actual);
+    }
+
+    public static function provideMatchingAssertions(): Generator
+    {
+        yield 'Expected Int64, Actual Int64' => [
+            'expected' => new Int64(8589934592),
+            'actual' => new Int64(8589934592),
+        ];
+
+        yield 'Expected Int64, Actual int' => [
+            'expected' => new Int64(8589934592),
+            'actual' => 8589934592,
```

Comment:
I realize we don't test PHPLIB on 32-bit platforms, but is this integer literal problematic or would PHP interpret it as a float?